### PR TITLE
Model api integration

### PR DESCRIPTION
**Pull request by IWorkForFood**

### CODE REVIEW

File: anynote/main/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        migrations.CreateModel(
+            name='NoteModel',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('content', models.JSONField(default=dict)),
+                ('hash', models.CharField(default='', max_length=200)),
+                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
+            ],
+        ),
+    ]
```
Comment (egorgur) [inline anynote/main/migrations/0001_initial.py]: You should not include migrations files in commits

File: anynote/api/views.py [hunk 2]
```diff
@@ -23,29 +32,108 @@ class UsersApiView(generics.ListCreateAPIView):
...
+            if user_id:
+                return User.objects.get(pk=user_id)
+            else:
+                raise Http404("Note not found.")
+
+        except jwt.exceptions.InvalidSignatureError:
+            raise PermissionDenied("Invalid access token.")
+
+        except NoteModel.DoesNotExist:
+            raise Http404("Note not found.")
```
Comment (egorgur) [inline anynote/api/views.py]: As we discussed, rename this UserApiView to AccountApiView and return old UserApiView from older commit. The endpoint users/<int:pk> will be used for testing purposes.

File: anynote/api/views.py [hunk 3]
```diff
@@ -23,29 +32,108 @@ class UsersApiView(generics.ListCreateAPIView):
...
 
 class UserApiView(generics.RetrieveUpdateDestroyAPIView):
     """
-    API endpoint that allows to get a specific ser data.
+    API endpoint that allows to get a specific user data.
 
     Only for debugging
     permission: is_staff
 
     """
```
Comment (egorgur) [inline anynote/api/views.py]: Edit this DockString. It must represent actual view functionality.

File: anynote/api/urls.py [hunk 4]
```diff
@@ -5,11 +5,16 @@
...
 urlpatterns = [
     path("users/", usersApiView, name="usersApi"),
-    path("users/<int:pk>/", userApiView, name="userApi"),
+    path("user/", userApiView, name="userApi"),
+
+    path("notes/", notesApiView, name="notesApi"),
+    path("note/", noteApiView, name="noteApi"),
 
     path("token/", tokenObtainPairView, name="token_obtain_pair"),
     path("token/refresh/", tokenRefreshView, name="token_refresh"),
```
Comment (egorgur) [inline anynote/api/urls.py]: Why you have deleted path("users/<int:pk>/") to individual users? It is used to debug user info.
Undo its deletion

File: anynote/main/migrations/0001_initial.py
Comment (egorgur) [inline anynote/main/migrations/0001_initial.py:1]: Do not include migrations files in branches and commits

**Review by egorgur — CHANGES_REQUESTED** [review]
Make changes as we discussed.

File: anynote/api/urls.py [hunk 5]
```diff
@@ -5,11 +5,18 @@
...
     tokenVerifyView,
+    notesApiView,
+    noteApiView,
+    accountApiView,
 )
 
 urlpatterns = [
     path("users/", usersApiView, name="usersApi"),
-    path("users/<int:pk>/", userApiView, name="userApi"),
+    path("user/<int:pk>", userApiView, name="userApi"),
```
Comment (egorgur) [inline anynote/api/urls.py]: Mistake in url naming. It should be user**s**/<int:pk>

**Review by egorgur — APPROVED** [review]
Great job, Michael! **<3**

File: anynote/api/serializers.py [hunk 6]
```diff
@@ -16,18 +17,38 @@ class UsersSerializer(serializers.ModelSerializer):
...
 
     class Meta:
         model = User
         fields = "__all__"  # ['id', 'username', 'last_login', 'date_joined', 'password']
+
+
+class NotesSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField()
+    content = serializers.JSONField(default=dict)
+    hash = serializers.CharField(max_length=200, default="")
```
Comment (mrFroxyk) [inline anynote/api/serializers.py:41]:
```suggestion
fdsfdsfsd
```

**Pull request merged**